lib/multiplexer/multiplexer.go

```diff
@@ -314,6 +340,8 @@ func detectProto(in []byte) (int, error) {
 		return ProtoSSH, nil
 	case bytes.HasPrefix(in, tlsPrefix):
 		return ProtoTLS, nil
+	case bytes.HasPrefix(in, httpPrefix):
```

Comment:
If the multiplexer is going to break out HTTP connections, then the other [verbs](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods) should also be supported.

Maybe something like below. GET is the first verb, so the performance should be the same since it will return from the if statement right away.

```go
func detectProto(in []byte) (int, error) {
	switch {
	// reader peeks only 3 bytes, slice the longer proxy prefix
	case bytes.HasPrefix(in, proxyPrefix[:3]):
		return ProtoProxy, nil
	case bytes.HasPrefix(in, sshPrefix):
		return ProtoSSH, nil
	case bytes.HasPrefix(in, tlsPrefix):
		return ProtoTLS, nil
	case isHTTP(in):
		return ProtoHTTP, nil
	default:
		return ProtoUnknown, trace.BadParameter("failed to detect protocol by prefix: %v", in)
	}
}

// isHTTP returns true if the prefix is for a HTTP method.
func isHTTP(in []byte) bool {
   if in == []byte("GET") || 
      in == []byte("POST") ||
      in == []byte("PUT") || 
      in == []byte("DELETE") ||
      in == []byte("HEAD") ||
      in == []byte("CONNECT") ||
      in == []byte("OPTIONS") ||
      in == []byte("TRACE") ||
      in == []byte("PATCH")  {
      return true
   }
   return false
}
```